src/gnssro/gnssro_bufr2ioda2.f90

```diff
@@ -0,0 +1,618 @@
+call check( nf90_put_att(grpid_metadata, varid_imph, "valid_range", "0 - 200 km" ))
+call check( nf90_def_var(grpid_metadata, "sensor_azimuth_angle", NF90_FLOAT, nlocs_dimid, varid_azim))
+call check( nf90_put_att(grpid_metadata, varid_azim, "longname", "GNSS->LEO line of sight" ))
+call check( nf90_put_att(grpid_metadata, varid_azim, "_FillValue", real(missingvalue) ))
+call check( nf90_put_att(grpid_metadata, varid_azim, "units", "Degree" ))
+call check( nf90_put_att(grpid_metadata, varid_azim, "valid_range", "0 - 360 degree" ))
+call check( nf90_def_var(grpid_metadata, "geoid_height_above_reference_ellipsoid",NF90_FLOAT, nlocs_dimid, varid_geoid))
+call check( nf90_put_att(grpid_metadata, varid_geoid, "longname", "Geoid height above WGS-84 ellipsoid" ))
+call check( nf90_put_att(grpid_metadata, varid_geoid, "units", "Meters" ))
+call check( nf90_put_att(grpid_metadata, varid_geoid, "valid_range", "-200 - 200 m" ))
+call check( nf90_def_var(grpid_metadata, "earth_radius_of_curvature",NF90_FLOAT, nlocs_dimid, varid_rfict))
+call check( nf90_put_att(grpid_metadata, varid_rfict, "longname", "Earth’s local radius of curvature" ))
+call check( nf90_put_att(grpid_metadata, varid_rfict, "units", "Meters" ))
+call check( nf90_put_att(grpid_metadata, varid_rfict, "valid_range", "6200 - 6600 km" ))
+call check( nf90_enddef(ncid) )
+
+call check( nf90_put_var(grpid_obsvalue, varid_ref, gpsro_data%ref(1:ndata)) )
+call check( nf90_put_var(grpid_obserror,varid_refoe,gpsro_data%refoe_gsi(1:ndata)) )
+call check( nf90_put_var(grpid_obsvalue, varid_bnd,gpsro_data%bend_ang(1:ndata)))
+call check( nf90_put_var(grpid_obserror,varid_bndoe,gpsro_data%bndoe_gsi(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_lat, gpsro_data%lat(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_lon, gpsro_data%lon(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_time, gpsro_data%time(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_datetime, gpsro_data%datetime(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_epochtime,gpsro_data%epochtime(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_recn, gpsro_data%recn(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_said, gpsro_data%said(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_siid, gpsro_data%siid(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_ptid, gpsro_data%ptid(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_sclf, gpsro_data%sclf(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_asce, gpsro_data%asce(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_ogce, gpsro_data%ogce(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_msl, gpsro_data%msl_alt(1:ndata)))
+call check( nf90_put_var(grpid_metadata, varid_impp,gpsro_data%impact_para(1:ndata)) )
+call check( nf90_put_var(grpid_metadata, varid_imph,gpsro_data%impact_para(1:ndata)-gpsro_data%rfict(1:ndata)-gpsro_data%geoid(1:ndata)) )
```

Comment:
```suggestion
call check( nf90_put_var(grpid_metadata, varid_imph,gpsro_data%impact_para(1:ndata) &
                                                   -gpsro_data%rfict(1:ndata) &
                                                   -gpsro_data%geoid(1:ndata)) )
```

Reply:
thank you Fabio for catching this. Have not tested with GNU.